new_course/1_programming_fundementals/problem_solving.md

```diff
@@ -1,31 +1,274 @@
+
+To gain clarity and understanding of the problem, write it down on paper, reword it in plain english until it makes sense to you, draw diagrams if that helps. When you can explain the problem to someone else in plain english, you understand it.
+
+### Plan
+Now that you know what you're aiming to solve, don't jump into coding just yet its time to plan out how you're going to solve it first.
+
+Some of the questions you should answer at this part of the process:
+
+* Does your program have a user interface? what will it look like? what functionality will the interface have? sketch this out on paper
+* What inputs will your program have? will the user enter data or will you get input from somewhere else
+* Whats the desired output?
+* Given your inputs, what are the steps necessary to return the desired output?
+
+The last question is where you will write out an algorithm to solve the problem. You can think of an algorithm as a recipe for solving a particular problem. It defines the steps that need to be taken by the computer to solve a problem in pseudo code.
+
+#### Pseudo Code
+Pseudo code is writing out the logic for your program in natural language instead of code. It helps you slow down and think through the steps your program will have to go through to solve the problem.
+
+Heres an example of what the pseudo code for a simple program that prints all numbers up to a inputted number might look like:
+
+```
+When the user inputs a number
+Initialize a counter variable and set its value to zero
+While counter is smaller than user inputted number
+Print the value of the counter variable
+```
+
+This is a very simple program to demonstrate how pseudo code looks, there will be more examples of pseudo code included in the assignments.
+
+
+### Divide and Conquer
+From your planning, you should have identified some sub problems of the big problem you’re solving. Each of the steps in the algorithm we wrote out in the last section are sub problems. Pick the smallest or simplest one and start there with coding.
+
+It's important to remember that you might not know all the steps that you might need up front so your algorithm may be incomplete, this is fine. Getting started with and solving one of the sub problems you have identified in the planning stage, often reveals the next sub problem you can work on. Or if you already know the next sub problem is it’s often simpler with the first sub problem solved.
+
+Many beginners try to solve the big problem in one go and. **Don’t do this**. If the problem is sufficiently complex enough you’ll get yourself tied in knots and make life a lot harder for yourself. Decomposing problems into smaller easier to solve sub problems is a much better approach as decomposition is the main way of dealing with complexity and making problems easier and more approachable to solve and understand.
```

Comment:
Many beginners try to solve the big problem in one go and. Remove the `and` at the end of the sentence.